server/base/src/test/java/org/apache/accumulo/server/manager/state/RootTabletStateStoreTest.java

```diff
@@ -139,8 +140,9 @@ public void testRootTabletStateStore() throws DistributedStoreException {
     assertThrows(IllegalArgumentException.class, () -> tstore.setFutureLocations(assignmentList));
 
     try {
-      TabletLocationState broken =
-          new TabletLocationState(notRoot, server, null, null, null, null, false);
+      TabletLocationState broken = new TabletLocationState(notRoot,
+          new TabletMetadata.Location(server, TabletMetadata.LocationType.FUTURE), null, null, null,
```

Comment:
```suggestion
          new TabletMetadata.Location.future(server), null, null, null,
```

Reply:
We don't have to but I could make the constructors protected/private so we don't miss instances of using the static methods like I missed this one.

Reply:
There are two constructors, would this mean 6 static methods are needed? Six static methods seems fine. Making the constructors private if it makes the code shorter and more consistent SGTM.

Reply:
Yeah, exactly, just to keep it consistent so we use the short static methods everywhere.

Reply:
Latest commit adds more static methods and makes the constructors private